File: apps/web/src/lib/post-status.ts
import type { PostStatus } from "@/lib/posts-api";

type Variant = "success" | "info" | "muted" | "warning";

export const POST_STATUS_META: Record<PostStatus, { label: string; variant: Variant }> = {
  DRAFT: { label: "초안", variant: "muted" },
  SCHEDULED: { label: "예약됨", variant: "info" },
  PUBLISHING: { label: "발행 중", variant: "warning" },
  PUBLISHED: { label: "발행됨", variant: "success" },
  PARTIAL: { label: "일부 발행", variant: "warning" },
  FAILED: { label: "실패", variant: "warning" },
  RECONNECT_REQUIRED: { label: "재연결 필요", variant: "warning" },
};

File: apps/api/src/main/java/com/postflow/post/PostStatus.java
package com.postflow.post;

public enum PostStatus {
    DRAFT,
    SCHEDULED,
    PUBLISHING,
    PUBLISHED,
    PARTIAL,            // 멀티채널 발행에서 일부만 성공(나머지 실패) — 부분 실패 허용
    FAILED,
    RECONNECT_REQUIRED,
    DELETED
}
